controllers/attester_controller.go

```diff
@@ -56,77 +59,204 @@ func (r *AttesterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	err := r.Get(ctx, req.NamespacedName, att)
 	if err != nil {
 		log.Error(err, "Unable to load attester")
-		return ctrl.Result{}, err
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	policy, err := attester.NewPolicy(req.Name, att.Spec.Policy, opaTrace)
-	if err != nil {
-		log.Error(err, "Unable to create policy")
-		att.Status.CompiledPolicy = "Failed to compile"
-		return ctrl.Result{}, err
-	}
+    // Register finalizer
+    err = r.registerFinalizer(log, att)
+    if err != nil {
+        log.Error(err, "Error registering finalizer")
+    }
 
-	// TODO: update status based on results of compiling the policy
-	att.Status.CompiledPolicy = "Compiled"
+    // If the attester is trying to be deleted remove the finalizer, delete the secret, 
+    // and remove the Attester object from r.Attesters
+    if !att.ObjectMeta.DeletionTimestamp.IsZero() && containsFinalizer(att.ObjectMeta.Finalizers, attesterFinalizerName) {
+        log.Info("Removing finalizer")
 
-	if err := r.Status().Update(ctx, att); err != nil {
-		log.Error(err, "Unable to update Attester status")
-		return ctrl.Result{}, err
-	}
+        // Removing finalizer
+        att.ObjectMeta.Finalizers = removeFinalizer(att.ObjectMeta.Finalizers, attesterFinalizerName)
+        err := r.Update(ctx, att)
+        if err != nil {
+            log.Error(err, "Error Removing the finalizer")
+            return ctrl.Result{}, err
+        }
+
+        // Deleting secret
+        secret := &corev1.Secret{}
+        err = r.Get(ctx, req.NamespacedName, secret)
+        if err != nil {
+            log.Error(err, "Failed to get the secret")
+        }
+
+        err = r.Delete(ctx, secret)
```

Comment:
this should be conditional...only delete if rode created the secret.  I'm thinking there should be an annotation you add to the secret if/when you created it to describe that rode owns it.  Then, only delete if the annotation is there.  This way if you manually create the secret, it isn't deleted when you delete the attester